File: JAVA/S1_BASIC/L7_OOPS/ENCAPSULATION/encapsulation.java

class Student {
    private int age;   // hidden data
    

    public int getAge() {
        return age;
    }

    public void setAge(int age) {
        if (age > 0) {
            this.age = age;
        }
    }
    
}
public class encapsulation{
    public static void main(String[] args){
        Student s=new Student();
        s.setAge(18);
        System.out.println(s.getAge());

    }
}

File: python/S1_BASIC/L7_OOPS/ENCAPSULATION/encapsulation.py
class Student:
    def __init__(self):
        self.__marks__=0
    def setmarks(self,m):
        if(m>=0):
            self.__marks=m
    def getmarks(self):
        return self.__marks
s=Student()
s.setage(20)
print(s.getage())
